Formula/lldb.rb

```diff
@@ -0,0 +1,75 @@
+class Lldb < Formula
+  desc "Next generation, high-performance debugger"
+  homepage "https://lldb.llvm.org"
+  url "https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.1/lldb-13.0.1.src.tar.xz"
+  sha256 "4a0c590bccbb3ce8fcbc1997aaed89a0ddbfd8be4b7f393230a867b2de62db16"
+  license "Apache-2.0" => { with: "LLVM-exception" }
+  head "https://github.com/llvm/llvm-project.git", branch: "main"
+
+  livecheck do
+    formula "llvm"
+  end
+
+  depends_on "cmake" => :build
+  depends_on "sphinx-doc" => :build
+  depends_on "swig" => :build
+  depends_on "llvm"
+  depends_on "python@3.10"
+  depends_on "six"
+  depends_on "xz"
+
+  uses_from_macos "libedit"
+  uses_from_macos "libxml2"
+  uses_from_macos "ncurses"
+  uses_from_macos "zlib"
+
+  on_macos do
+    keg_only "provided_by_macos" # TODO: change to :provided_by_macos
+  end
+
+  on_linux do
+    depends_on "gcc"
+  end
+
+  fails_with gcc: "5"
+
+  def python_executable
+    Formula["python@3.10"].opt_bin/"python3"
+  end
+
+  # we install the lldb Python module into libexec to prevent users from
+  # accidentally importing it with a non-Homebrew Python or a Homebrew Python
+  # in a non-default prefix. See https://lldb.llvm.org/resources/caveats.html
+  def site_packages
+    Language::Python.site_packages(python_executable).delete_prefix("lib/")
+  end
+
+  def install
+    cd "lldb" if build.head?
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args,
+                    "-DCLANG_LINK_CLANG_DYLIB=ON",
+                    "-DLLVM_LINK_LLVM_DYLIB=ON",
```

Comment:
`CLANG_LINK_CLANG_DYLIB` should only affect the `clang` (and related tools) build, so it should have no effect on `lldb`.

We want to use `LLVM_LINK_LLVM_DYLIB` so that `lldb` and `liblldb` link with `libLLVM` instead of carrying around a private copy.

`brew` manages the `RPATH`s just fine so we can probably either patch this out or just remove it ourselves some other way.

Reply:
It looks like `CLANG_LINK_CLANG_DYLIB` embeds linkage, but not sure how it is used. Haven't had chance to look through

Related diff: https://reviews.llvm.org/D68456

Related cmake: https://github.com/llvm/llvm-project/blob/main/lldb/cmake/modules/AddLLDB.cmake#L97-L101
```cmake
    if(CLANG_LINK_CLANG_DYLIB)
      target_link_libraries(${name} PRIVATE clang-cpp)
    else()
      target_link_libraries(${name} PRIVATE ${PARAM_CLANG_LIBS})
    endif()
```